mlir/include/mlir/Dialect/OpenMP/OpenMPOps.td

```diff
@@ -1198,6 +1199,9 @@ def MapInfoOp : OpenMP_Op<"map_info", [AttrSizedOperandSegments]> {
     - 'map_capture_type': Capture type for the variable e.g. this, byref, byvalue, byvla
        this can affect how the variable is lowered.
     - `name`: Holds the name of variable as specified in user clause (including bounds).
+    -  'is_fortran_allocatable': Indicates if the var_ptr variable is a fortran allocatable 
+        type, e.g. a pointer or allocatable containing a descriptor mapping that wraps the 
+        data and contains further information on the mapped variable.
```

Comment:
I'll do my best to explain some of the choices made in this PR after I update the PR with the requested changes, I'll add a "ping" message when it's updated.

As for the RFC requested in another message, I'm more than happy to do so, but it might be worth discussing the choices in the PR first before we commit to an RFC, in-case any reviewer can perhaps see alternatives to the choices made after I explain them a little more. But perhaps it's just better to go ahead and open an RFC, if you think that's a better approach then I'm happy to do an RFC alongside the discussion in the patch? As I guess that's still a very good way to get some feedback.